test/integration/gardenlet/vpaevictionrequirements/vpaevictionrequirements_test.go

```diff
@@ -0,0 +1,211 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpaevictionrequirements_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+)
+
+var _ = Describe("VPA EvictionRequirements controller tests", func() {
+	var (
+		vpa                            *vpaautoscalingv1.VerticalPodAutoscaler
+		targetDeployment               *appsv1.Deployment
+		maintenanceWindowNow           *gardencorev1beta1.MaintenanceTimeWindow
+		maintenanceWindowAlreadyPassed *gardencorev1beta1.MaintenanceTimeWindow
+		upscaleOnlyRequirement         = []*vpaautoscalingv1.EvictionRequirement{{
+			Resources:         []corev1.ResourceName{corev1.ResourceMemory, corev1.ResourceCPU},
+			ChangeRequirement: vpaautoscalingv1.TargetHigherThanRequests,
+		}}
+	)
+
+	BeforeEach(func() {
+		fakeClock.SetTime(time.Now())
+
+		maintenanceWindowNow = &gardencorev1beta1.MaintenanceTimeWindow{
+			Begin: time.Now().Format("150405-0700"),
+			End:   time.Now().Add(1 * time.Hour).Format("150405-0700"),
+		}
+		maintenanceWindowAlreadyPassed = &gardencorev1beta1.MaintenanceTimeWindow{
+			Begin: time.Now().Add(-3 * time.Hour).Format("150405-0700"),
+			End:   time.Now().Add(-2 * time.Hour).Format("150405-0700"),
+		}
+
+		targetDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-deployment",
+				Namespace: testNamespace.Name,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "test"}},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "test"}},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{Name: "test-container", Image: "my-nonexisting-image"},
+						},
+					},
+				},
+			},
+		}
+
+		vpa = &vpaautoscalingv1.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-vpa",
+				Namespace: testNamespace.Name,
+				Labels: map[string]string{
+					testID: testRunID,
+					v1beta1constants.LabelVPAEvictionRequirementsController: v1beta1constants.EvictionRequirementManagedByController,
+				},
+			},
+			Spec: vpaautoscalingv1.VerticalPodAutoscalerSpec{
+				TargetRef:    &autoscalingv1.CrossVersionObjectReference{Name: targetDeployment.Name, APIVersion: "apps/v1", Kind: "Deployment"},
+				UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{EvictionRequirements: []*vpaautoscalingv1.EvictionRequirement{}},
```

Comment:
```suggestion
				UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{EvictionRequirements: nil},
```

sounds like a more real-world scenario.

Reply:
done.